draft-kazuho-quic-authenticated-handshake.md

```diff
@@ -318,26 +318,25 @@ packet means that the retransmitted Initial packets would become undecryptable
 and therefore be deemed lost by the client, reducing the client's congestion
 window size.
 
-## No Support for Split Mode
-
-Under the design discussed in this document, it is impossible to use an
-unmodified QUIC server as a backend server in "Split Mode" ([TLS-ESNI];
-section 3) due to the following two reasons:
-
-* Access to initial_auth_secret is required for generating and validating
-  Initial packets.  However, the backend server, not knowing the ESNI private
-  key, cannot calculate the secret.
-
-* The client-facing server cannot continue forwarding packets to the correct
-  destination when there is a change in Connection ID mid-connection.
-
-To address the issues, we might consider specifying a protocol that will be
-used between the client-facing server and the backend server for communicating
-the initial_auth_secret and the spare Connection IDs.  Note that such protocol
-can be lightweight, assuming the communication between the two servers will be
-over a virtual private network.  Such assumption can be made because the
-backend server cannot operate QUIC without access to the source address-port
-tuple of the packets that the client has sent.
+## Split Mode
+
+To support server-side deployments using "Split Mode" ([TLS-ESNI]; section 3),
+the following properties need to be exchanged between the fronting server and
+the hidden server, in addition to those required by the Encrypted SNI
+extension:
+
+* hmac_key
+* spare Connection IDs
+
```

Comment:
I think the fronting server also needs to provide the client with the source address of the incoming Client Initial, and also with the ODCID in case of retry.

Reply:
Nice catch!

Added ODCID in 51dd707.

I'd add another commit covering the communication of source address; it is not only necessary for Initial but for every packet, because the hidden server needs to validate the path when the client migrates to a new address.

Reply:
Did you have time to write the part about providing the source address, or should we just check the PR as is?